cdap-ui/app/cdap/components/DataPrep/DataPrepServiceControl/index.js

```diff
@@ -23,32 +23,37 @@ import classnames from 'classnames';
 import MyDataPrepApi from 'api/dataprep';
 import {i18nPrefix, MIN_DATAPREP_VERSION, artifactName} from 'components/DataPrep';
 import isObject from 'lodash/isObject';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { PROGRAM_STATUSES } from 'services/global-constants';
+import { objectQuery } from 'services/helpers';
 
 require('./DataPrepServiceControl.scss');
 
 const PREFIX = `features.DataPrepServiceControl`;
 
 export default class DataPrepServiceControl extends Component {
-  constructor(props) {
-    super(props);
+  state = {
+    loading: false,
+    error: null,
+    extendedMessage: null
+  };
 
-    this.state = {
-      loading: false,
-      error: null,
-      extendedMessage: null
-    };
-
-    this.enableService = this.enableService.bind(this);
-  }
-
-  componentWillUnmount() {
-    if (this.servicePoll && this.servicePoll.unsubscribe) {
-      this.servicePoll.unsubscribe();
-    }
+  checkForServiceStatusBeforeStart = () => {
```

Comment:
should this functionality just be in the ServiceEnabler service? Also this check for serviceStatus is not accurate, we need to do ping one of the actual service methods to check whether it is running or not. If it is not available then we need to check for whether service is running or not, if it is, then wait until the ping is working.

Reply:
Originally had it in dataprep to avoid making changes that can impact other apps using the `ServiceEnablerUtilities`. But moved it anyways and modified other apis to have the same function for compatibility.